### PR TITLE
Add provider layer and hide constructor

### PR DESCRIPTION
**Pull request by nbramblett**

### CODE REVIEW

File: src/CoreProvider.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ */
+export function ProvideChatCore(config: ChatConfig): ChatCore {
+  return new ChatCore(config);
+}
+
+/**
+ * Provider for the ChatCore library with additional internal-only configuration
+ * @internal
+ */
+export function ProvideChatCoreInternal(
```
Comment (yen-tt) [inline src/CoreProvider.ts:16]: nit: functions in JS usually don't lead with uppercase letter. can we update this and above to be `provideChatCore`?

Also, we can overload the function by naming them the same and only define one signature

File: etc/chat-core.api.md [hunk 2]
```diff
@@ -17,12 +17,10 @@ export interface ChatConfig {
     version?: "STAGING" | "PRODUCTION" | number;
 }
 
-// @public
-export class ChatCore {
-    constructor(chatConfig: ChatConfig, promptPackage?: ChatPrompt);
-    getNextMessage(request: MessageRequest): Promise<MessageResponse>;
-    streamNextMessage(request: MessageRequest): Promise<StreamResponse>;
-}
+// Warning: (ae-forgotten-export) The symbol "ChatCore_2" needs to be exported by the entry point index.d.ts
```
Comment (yen-tt) [inline etc/chat-core.api.md:20]: seems like the class needs to be exported anyway

File: etc/chat-core.api.md [hunk 3]
```diff
@@ -50,6 +48,14 @@ export enum Environment {
     SANDBOX = "SANDBOX"
 }
 
+// Warning: (ae-internal-missing-underscore) The name "InternalConfig" should be prefixed with an underscore because the declaration is marked as @internal
```
Comment (yen-tt) [inline etc/chat-core.api.md:51]: not sure if we want to follow this rule -- If not, lets update api extractor to ignore this rule

File: src/CoreProvider.ts (hunk 1 above)
Comment (nbramblett) [reply src/CoreProvider.ts:16]: Js can't be overloaded and I think we want them separate for @internal

File: etc/chat-core.api.md (hunk 2 above)
Comment (nbramblett) [reply etc/chat-core.api.md:20]: rip, updating

File: etc/chat-core.api.md (hunk 3 above)
Comment (nbramblett) [reply etc/chat-core.api.md:51]: Yeah I'm not a fan of it for interfaces, that seems unnecessary. Updating.

File: src/ChatCore.ts [hunk 4]
```diff
@@ -23,12 +28,12 @@ export class ChatCore {
   private endpoints: Endpoints;
   private promptPackage: ChatPrompt;
```
Comment (yen-tt) [inline src/ChatCore.ts:30]: can we update the doc for this class -- it's not the entry point of the library anymore, the provider layer is. and include a `@remarks` tag that this is consider internal and to encourage user to use the provider layer

File: src/index.ts [hunk 5]
```diff
@@ -1,3 +1,7 @@
 export { ChatCore } from "./ChatCore";
 export { StreamResponse } from "./infra/StreamResponse";
+export {
+  provideChatCore as provideChatCore,
```
Comment (yen-tt) [inline src/index.ts:4]: remove `as provideChatCore`

**Pull request merged**